test/infrastructure/docker/e2e/docker_suite_test.go

```diff
@@ -175,3 +187,74 @@ func writeLogs(mgmt *CAPDCluster, namespace, deploymentName, logDir string) erro
 	}
 	return nil
 }
+
+func dumpResources(mgmt *CAPDCluster, resourcePath string) error {
```

Comment:
You can also potentially wrap the `kubectl cluster-info dump` command rather than having to build out this logic separately.